pandas/_libs/algos.pyx

```diff
@@ -65,22 +65,22 @@ class Infinity(object):
     """ provide a positive Infinity comparision method for ranking """
 
     __lt__ = lambda self, other: False
-    __le__ = lambda self, other: self is other
-    __eq__ = lambda self, other: self is other
-    __ne__ = lambda self, other: self is not other
-    __gt__ = lambda self, other: self is not other
+    __le__ = lambda self, other: isinstance(other, Infinity)
```

Comment:
can you add some tests for Infinity/NegInfitiy (some in test_algos already). esp how they react with NaN

Reply:
Yes, I can add some tests. 
Is it necessary to take the NaN into consideration, since Infinity/NegInfitiy is used to replace all NaN values ? There are no NaNs when comparing elements against each other. 
```
np.putmask(values, mask, nan_value)
```
nan_value is eiterh Infinity or NegInfitiy